Sake/Mixer.cs

```diff
@@ -291,31 +291,51 @@ public IEnumerable<ulong> Decompose(IEnumerable<Money> myInputsParam, IEnumerabl
 
             setCandidates.Add( 
                 hash.ToHashCode(), // Create hash to ensure uniqueness.
-                (naiveSet.Select(x => (ulong)x.EffectiveCost.Satoshi), loss + (ulong)naiveSet.Count * OutputFee + (ulong)naiveSet.Count * InputFee)); // The cost is the remaining + output cost + input cost.
+                (naiveSet.Select(x => (ulong)x.EffectiveCost.Satoshi), loss + CalculateCostMetrics(naiveSet)));
 
 
             // Create many decompositions for optimization.
             var stdDenoms = denoms.Select(x => x.EffectiveCost.Satoshi).Where(x => x <= myInputSum).Select(x => (long)x).ToArray();
-            var maxNumberOfOutputsAllowed = (int)Math.Min(availableVsize / InputSize, 8); // The absolute max possible with the smallest script type.
+            var smallestScriptType = Math.Min(ScriptType.P2WPKH.EstimateOutputVsize(), ScriptType.Taproot.EstimateOutputVsize());
+            var maxNumberOfOutputsAllowed = Math.Min(availableVsize / smallestScriptType, 8); // The absolute max possible with the smallest script type.
             var tolerance = (long)Math.Max(loss, 0.5 * MinAllowedOutputAmountPlusFee); // Taking the changefee here, might be incorrect however it is just a tolerance.
 
-            foreach (var (sum, count, decomp) in Decomposer.Decompose(
-                target: (long)myInputSum,
-                tolerance: tolerance,
-                maxCount: Math.Min(maxNumberOfOutputsAllowed, 8),
-                stdDenoms: stdDenoms))
+            if (maxNumberOfOutputsAllowed > 1)
```

Comment:
What's this?

Reply:
What if it isn't? How can it be that it isn't?

Reply:
If availableVsize is small. It is unlikely but it is coming from the coordinator credential issuer. Maybe it is abnormal and we should bail out the round. But to keep the scope on this specific code. It makes no sense to start looking for combinations if I can only register one output.

Reply:
BTW the master contains this condition, Lucas added it long ago.

Reply:
@lontivero if you make modifications to this file in the master, please make sure to keep Sake in sync